spark/src/main/java/org/opensearch/sql/spark/dispatcher/SparkQueryDispatcher.java

```diff
@@ -106,4 +129,80 @@ private String constructSparkParameters(String datasourceName) throws URISyntaxE
         "spark.sql.catalog." + datasourceName, FLINT_DELEGATE_CATALOG);
     return s3GlueSparkSubmitParameters.toString();
   }
+
+  private StartJobRequest getStartJobRequestForNonIndexQueries(
+      DispatchQueryRequest dispatchQueryRequest) {
+    StartJobRequest startJobRequest;
+    FullyQualifiedTableName fullyQualifiedTableName =
+        SQLQueryUtils.extractFullyQualifiedTableName(dispatchQueryRequest.getQuery());
+    if (fullyQualifiedTableName == null || fullyQualifiedTableName.getDatasourceName() == null) {
+      throw new UnsupportedOperationException("Queries without a datasource are not supported");
+    }
+    dataSourceUserAuthorizationHelper.authorizeDataSource(
+        this.dataSourceService.getRawDataSourceMetadata(
+            fullyQualifiedTableName.getDatasourceName()));
+    String jobName =
```

Comment:
Query type will decide job name? We could refactor the code later.

Reply:
Yes.